### PR TITLE
rPackages.immunotation: cache external URLs

### PR DESCRIPTION
**Pull request by Kupac**
## Description of changes

like before: https://github.com/NixOS/nixpkgs/pull/303724

Immunotation uses 3 external URL-s, and stores them variables in the package's namespace. Package installation is not possible without internet connection, and skipping load test doesn't help, because this happens before the load test.

I use fetchurl to get those external deps, and replace them in the code with a patch + substituteInPlace.

This should work in the long term for the two MHC allele lists, The geographical region list needs to be taken from an archive.org snapshot of the site, because the source site changes a bit every day. The snapshot can be updated eventually. Let's say each year or two, or when something breaks. But the regions/countries/ethnicities don't change so often.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Kupac** [comment]
@ofborg build rPackages.immunotation

**Review by b-rodrigues — APPROVED** [review]
builds on aarch64-linux, LGTM!

File: pkgs/development/r-modules/default.nix [hunk 1]
```diff
@@ -1318,6 +1318,43 @@ let
...
+    immunotation = let
+      MHC41alleleList = fetchurl {
+        url = "https://services.healthtech.dtu.dk/services/NetMHCpan-4.1/allele.list";
+        hash = "sha256-CRZ+0uHzcq5zK5eONucAChXIXO8tnq5sSEAS80Z7jhg=";
+      };
+
+      MHCII40alleleList = fetchurl {
+        url = "https://services.healthtech.dtu.dk/services/NetMHCIIpan-4.0/alleles_name.list";
+        hash = "sha256-K4Ic2NUs3P4IkvOODwZ0c4Yh8caex5Ih0uO5jXRHp40=";
+      };
```
Comment (jbedo) [inline pkgs/development/r-modules/default.nix:1331]: Are these stable or should they be archived on the web archive?

Comment (Kupac) [reply pkgs/development/r-modules/default.nix:1331]: The database is not updated very frequently, last update was in 2020. But even when they do update it, you can still access the previous versions. For example, https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/alleles_name.list, from 2018. So as long as the server is alive, I don't see a need to point to web archive.

**Review by jbedo — APPROVED** [review]

**Pull request merged**